x-pack/plugins/canvas/public/components/workpad_loader/index.js

```diff
@@ -34,6 +44,9 @@ export const WorkpadLoader = compose(
           props.router.navigateTo('loadWorkpad', { id: workpad.id, page: 1 });
         } catch (err) {
           notify.error(err, { title: `Couldn't upload workpad` });
+          // TODO: remove this and switch to checking user privileges when canvas loads when granular app privileges are introduced
+          // https://github.com/elastic/kibana/issues/20277
+          if (err.response.status === 403) props.setCanUserWrite(false);
```

Comment:
I also added a few of these in here since you last reviewed the PR. I added the same check for the create, clone, and remove workpad handlers.

Reply:
Perfect.